src/components/Camera/StandardCamera/StandardCamera.js

```diff
@@ -100,6 +105,19 @@ const StandardCamera = ( {
 
   const { screenWidth } = useDeviceOrientation( );
 
+  const showStorageFullAlert = () => Alert.alert(
+    "Device Storage Full",
+    "iNaturalist may not be able to save your photos or may crash",
+    [{ text: t( "OK" ) }]
+  );
+
+  DeviceInfo.getFreeDiskStorage().then( freeDiskStorage => {
+    // considered full when 200MB left
+    if ( freeDiskStorage <= MIN_DEVICE_STORAGE ) {
+      setDeviceStorageFull( true );
+    }
+  } );
+
```

Comment:
Don't repeat yourself. This is the 4th time you've had almost identical code showing this modal, and the third time you're doing the same check with `DeviceInfo.getFreeDiskStorage()`. If you do the same thing twice, it's time to pull that logic out into a module (seems like a hook would do the trick here). If there are small tweaks to how it has to behave, like making the alert have a cancel button or not, those can be options passed into whatever module you create.